src/project-manager.ts

```diff
@@ -735,6 +754,11 @@ export class ProjectConfiguration {
 	private expectedFilePaths = new Set<string>();
 
 	/**
+	 * List of resolved extra root directories to allow global type declaration files to be loaded from.
+	 */
+	private typeRoots: string[];
```

Comment:
is the order important? if not, use a Set

Reply:
Not sure if `this.typeRoots.some(root => fileName.startsWith(root));` is a good fit for Set?